File: 136-single-number.py
# look at js file for thought process

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        nums.sort()
        print(nums)

        for i in range(0, len(nums), 2):
            if i == len(nums) - 1:
                return nums[i]
            if nums[i] != nums[i+1]:
                return nums[i]

File: 136-single-number.js
/*
Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.

You must implement a solution with a linear runtime complexity and use only constant extra space.

Example 1:

Input: nums = [2,2,1]
Output: 1

Example 2:

Input: nums = [4,1,2,1,2]
Output: 4

Example 3:

Input: nums = [1]
Output: 1
*/

/*
1.) sort nums
2.) iterate over sorted nums for even indices
3.) if the next number isn't the same as the current number, it's the single
4.) return that number
*/

var singleNumber = function(nums) {
    nums.sort();

    for (let i = 0; i < nums.length; i+=2) {
        if (nums[i] !== nums[i+1]) return nums[i];
    }
};
